src/Mod/Part/App/Attacher.cpp

```diff
@@ -1177,21 +1179,34 @@ Base::Placement AttachEngine3D::calculateAttachedPlacement(const Base::Placement
         if (vertex.IsNull())
             throw Base::ValueError("Null vertex in AttachEngine3D::calculateAttachedPlacement()!");
 
-        BRepAdaptor_Surface surf (face);
         Handle (Geom_Surface) hSurf = BRep_Tool::Surface(face);
         gp_Pnt p = BRep_Tool::Pnt(vertex);
 
         GeomAPI_ProjectPointOnSurf projector(p, hSurf);
         double u, v;
         if (projector.NbPoints()==0)
             throw Base::ValueError("AttachEngine3D::calculateAttachedPlacement: projecting point onto surface failed.");
+
         projector.LowerDistanceParameters(u, v);
+        GeomLProp_SLProps prop(hSurf, u, v, 1, Precision::Confusion());
+        gp_Dir dirX;
+        Standard_Boolean done;
 
-        BRepLProp_SLProps prop(surf,u,v,1, Precision::Confusion());
```

Comment:
Be careful here! There is a difference between shapes and geometries. The former can have a placement (or in OCC jargon TopLoc_Location) or orientation and both affect the normal vector.

Example:
```
f=Part.makePlane(1,1)
f.normalAt(0,0) # => (0,0,1)
f.Surface.normal(0,0) # => (0,0,1)
f.reverse()
f.normalAt(0,0) # => (0,0,-1)
f.Surface.normal(0,0) # => (0,0,1)

f.Placement.Rotation.Angle=1
f.Placement.Rotation.Axis=(1,1,1)
f.normalAt(0,0) # => (-0.639, 0.332, -0.693)
f.Surface.normal(0,0) # => (0.639, -0.332, 0.693)
```

The last line needs some more explanation. When running `f.Surface` then internally BRepAdaptor_Surface is used that applies the transformation of the placement to the returned surface. That's why the normal vector is not (0,0,1).
However, when accessing the original surface with `BRep_Tool::Surface` then you will still get the normal (0,0,1).

Reply:
Thanks for the aclaration. If I understand correctly, `BRep_Tool::Surface(const TopoDS_Face&, TopLoc_Location& L)` returns the surface without the tranformation applied but the overload `BRep_Tool::Surface(const TopoDS_Face&)` returns a copy of the surface with the transformation applied. The difference I see between working with the `BRepAdaptor_Surface` and the `Handle(Geom_Surface)` returned by `BRep_Tool::Surface(const TopoDS_Face&)` is the restriction on the parameters performed by `BRepAdaptor_Surface`. Is there any error in these statements? What is the impact of this restriction?